New maintainer guide (#4472)
jenkinsci/jenkins: .github/PULL_REQUEST_TEMPLATE.md

### Submitter checklist

> - [ ] Jira issue is well described (not mandatory)

Reviewer: Ambiguous:

* Is it not necessary to have a Jira issue?
* Is it not necessary that the Jira issue is useful?

Author: What if I adjust it like...

```suggestion
- [ ] Jira issue is well described (if needed)
```

Can just revert it tho

Reviewer: How about

```suggestion
- [ ] (If applicable) Jira issue is well described
```

- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))

> - [ ] If it would make sense to backport the change to LTS, a Jira issue must exist and be labeled as `lts-candidate` to be considered

Reviewer: ```suggestion
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).
```

- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))

> - [ ] If it would make sense to backport the change to LTS, a Jira issue should exist and be labeled as `lts-candidate`

Reviewer 2: If its required to backport maybe the wording should be clear here.
```suggestion
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist and be labeled as `lts-candidate` to be considered
```
